--- native/Programs/python.c	2022-09-11 12:23:30.000000000 -0700
+++ wasm/Programs/python.c	2022-10-19 10:26:04.000000000 -0700
@@ -2,6 +2,11 @@
 
 #include "Python.h"
 
+// This C file, which is defined right now in the dylink nodejs module, adds
+// the declaration that make everything extension modules need
+// accessible from outside our "python.wasm as a shared library".
+#include "libpython.c"
+
 #ifdef MS_WINDOWS
 int
 wmain(int argc, wchar_t **argv)
@@ -9,6 +14,11 @@
     return Py_Main(argc, argv);
 }
 #else
+
+// Make the main symbol visible.  We do this explicitly since our trick with #define main
+// doesn't work with python, due to there being other code that uses main as a struct attribute.
+
+__attribute__((visibility("default")))
 int
 main(int argc, char **argv)
 {
