--- native/Lib/pydoc.py	2022-10-24 10:35:39
+++ wasm/Lib/pydoc.py	2022-11-06 07:29:57
@@ -1565,6 +1565,8 @@
         return plainpager
     if sys.platform == "emscripten":
         return plainpager
+    if sys.platform == "wasi":
+        return ttypager
     use_pager = os.environ.get('MANPAGER') or os.environ.get('PAGER')
     if use_pager:
         if sys.platform == 'win32': # pipes completely broken in Windows
