--- native/Tools/wasm/wasm_assets.py	2022-10-24 10:35:39
+++ wasm/Tools/wasm/wasm_assets.py	2022-11-09 13:17:41
@@ -89,21 +89,24 @@
     "wsgiref/",
 )
 
+# The module detection code isn't robust, so I manually comment out
+# some of these from the list, which causes the corresponding py files
+# to NOT be excluded.
 OMIT_MODULE_FILES = {
     "_asyncio": ["asyncio/"],
     "audioop": ["aifc.py", "sunau.py", "wave.py"],
     "_crypt": ["crypt.py"],
-    "_curses": ["curses/"],
+    #"_curses": ["curses/"],
     "_ctypes": ["ctypes/"],
-    "_decimal": ["decimal.py"],
+    #"_decimal": ["decimal.py"],
     "_dbm": ["dbm/ndbm.py"],
     "_gdbm": ["dbm/gnu.py"],
-    "_json": ["json/"],
+    #"_json": ["json/"],
     "_multiprocessing": ["concurrent/", "multiprocessing/"],
     "pyexpat": ["xml/", "xmlrpc/"],
-    "readline": ["rlcompleter.py"],
-    "_sqlite3": ["sqlite3/"],
-    "_ssl": ["ssl.py"],
+    #"readline": ["rlcompleter.py"],
+    #"_sqlite3": ["sqlite3/"],
+    #"_ssl": ["ssl.py"],
     "_tkinter": ["idlelib/", "tkinter/", "turtle.py", "turtledemo/"],
 
     "_zoneinfo": ["zoneinfo/"],
@@ -127,9 +130,15 @@
 def get_sysconfigdata(args: argparse.Namespace) -> pathlib.Path:
     """Get path to sysconfigdata relative to build root
     """
-    data_name = sysconfig._get_sysconfigdata_name()
-    assert "emscripten_wasm32" in data_name
-    filename = data_name + ".py"
+    # python-wasm: Using sysconfig does NOT work for us, because
+    # we simply can't run our python.wasm under wasmtime (say), since too
+    # much is missing, but that's what is done in the Makefile.  So
+    # we comment this out, and explicitly put the path back in, like
+    # was done in 3.11.beta4.  Also, wasmtime is not a dependency of CoWasm.
+    # data_name = sysconfig._get_sysconfigdata_name()
+    # assert "emscripten_wasm32" in data_name
+    # filename = data_name + ".py"
+    filename = "_sysconfigdata__wasi_wasm32-wasi.py"
     return args.builddir / filename
 
 
