--- native/Python/fileutils.c	2022-09-11 12:23:30.000000000 -0700
+++ wasm/Python/fileutils.c	2022-09-18 11:58:42.000000000 -0700
@@ -1323,11 +1323,25 @@
     return get_inheritable(fd, 1);
 }

+extern int python_wasm_set_inheritable(int fd, int inheritable);

 /* This function MUST be kept async-signal-safe on POSIX when raise=0. */
 static int
 set_inheritable(int fd, int inheritable, int raise, int *atomic_flag_works)
 {
+    // For python-wasm we *have* to write this ourselves, since fcntl is
+    // built into upstream WASI and it makes this sort of stuff a no-op, or
+    // at least no functions in wasi get called.  Note that this has limited
+    // impact, e.g., it can only work in a WASI context for certain types of
+    // fd's, e.g., pipes.
+    int result = python_wasm_set_inheritable(fd, inheritable);
+    if (result < 0) {
+        if (raise)
+            PyErr_SetFromErrno(PyExc_OSError);
+        return -1;
+    }
+    return 0;
+
 #ifdef MS_WINDOWS
     HANDLE handle;
     DWORD flags;
