--- native/Lib/threading.py	2022-10-24 10:35:39
+++ wasm/Lib/threading.py	2022-11-15 09:41:01
@@ -945,6 +945,13 @@
         same thread object.
 
         """
+        # CoWasm -- if somehow something tries to actually start a thread, we just
+        # don't. CoWasm has no support for threads, but Python packages
+        # such as pip just assume it does.  E.g., pip's progress bar uses threading,
+        # and showing this message is better than crashing pip.
+        print("(WARNING: threading is not supported)")
+        return
+
         if not self._initialized:
             raise RuntimeError("thread.__init__() not called")
 
@@ -1101,6 +1108,8 @@
         exception.
 
         """
+        # CoWasm: we're making the thread a no-op, so it's done.
+        return
         if not self._initialized:
             raise RuntimeError("Thread.__init__() not called")
         if not self._started.is_set():
