--- native/numpy/core/setup.py	2022-08-13 11:30:12.808497700 -0700
+++ wasm/numpy/core/setup.py	2022-09-09 16:25:46.943622000 -0700
@@ -195,7 +195,7 @@
     check_funcs(OPTIONAL_STDFUNCS)
     check_funcs(OPTIONAL_FILE_FUNCS, headers=["feature_detection_stdio.h"])
     check_funcs(OPTIONAL_MISC_FUNCS, headers=["feature_detection_misc.h"])
-    
+
 
 
     for h in OPTIONAL_HEADERS:
@@ -739,6 +739,8 @@
         ):
             is_cpp = lang == 'c++'
             if is_cpp:
+                continue # TODO: for wasm for now
+
                 # this a workaround to get rid of invalid c++ flags
                 # without doing big changes to config.
                 # c tested first, compiler should be here
