--- native/numpy/distutils/command/build_ext.py	2022-09-09 06:36:30.000000000 -0700
+++ wasm/numpy/distutils/command/build_ext.py	2022-09-19 20:32:04.000000000 -0700
@@ -510,6 +510,11 @@
                 **kws)
         if cxx_sources:
             log.info("compiling C++ sources")
+            # python-wasm: the heuristics that numpy uses to
+            # convert the c compiler to a c++ compiler don't work
+            # for us since we use "cowasm-cc", so we manually
+            # make the change here:
+            cxx_compiler.compiler_so[1] = 'c++'
             c_objects += cxx_compiler.compile(
                 cxx_sources,
                 output_dir=output_dir,
