diff --git a/files.h b/files.h
--- a/files.h
+++ b/files.h
@@ -100,10 +100,6 @@ namespace files {
  void printHeckeElt(FILE* file, const H& h, const SchubertContext& p,
                             const Interface& I, OutputTraits& traits,
                             const Length& l = undef_length);
-template <class H>
-  void printHeckeElt(FILE* file, const H& h, const SchubertContext& p,
-			     const Interface& I, OutputTraits& traits,
-			     const Length& l = undef_length);
 template <class H>
   void printHeckeElt(FILE* file, const H& h, const Permutation& a,
                             const SchubertContext& p, const Interface& I,
