--- native/src/main.c	2022-10-01 09:10:53.000000000 -0700
+++ wasm/src/main.c	2022-10-04 08:18:26.000000000 -0700
@@ -76,7 +76,7 @@
 STATIC void read_profile(const char *);
 STATIC char *find_dot_file(char *);
 static int cmdloop(int);
-int main(int, char **);
+int dash_main(int, char **);
 
 /*
  * Main routine.  We initialize things, parse the arguments, execute
@@ -87,7 +87,7 @@
  */
 
 int
-main(int argc, char **argv)
+dash_main(int argc, char **argv)
 {
 	char *shinit;
 	volatile int state;
@@ -188,7 +188,6 @@
 	/* NOTREACHED */
 }
 
-
 /*
  * Read and execute commands.  "Top" is nonzero for the top level command
  * loop; it turns on prompting if the shell is interactive.
