--- native/Lib/_pyrepl/pager.py	2026-06-11 00:00:00.000000000 -0700
+++ wasm/Lib/_pyrepl/pager.py	2026-06-11 00:00:00.000000000 -0700
@@ -21,6 +21,8 @@
         return plain_pager
     if sys.platform == "emscripten":
         return plain_pager
+    if sys.platform == "wasi":
+        return tty_pager
     use_pager = os.environ.get('MANPAGER') or os.environ.get('PAGER')
     if use_pager:
         if sys.platform == 'win32': # pipes completely broken in Windows
@@ -75,7 +77,7 @@
         def getchar() -> str:
             return sys.stdin.read(1)
 
-    except (ImportError, AttributeError, io.UnsupportedOperation):
+    except (termios.error, ImportError, AttributeError, io.UnsupportedOperation):
         def getchar() -> str:
             return sys.stdin.readline()[:-1][:1]
 
