--- native/Lib/subprocess.py
+++ wasm/Lib/subprocess.py
@@ -75,7 +75,7 @@
     _mswindows = True
 
 # some platforms do not support subprocesses
-_can_fork_exec = sys.platform not in {"emscripten", "wasi", "ios", "tvos", "watchos"}
+_can_fork_exec = sys.platform not in {"emscripten", "ios", "tvos", "watchos"}
 
 if _mswindows:
     import _winapi
--- native/Modules/_posixsubprocess.c
+++ wasm/Modules/_posixsubprocess.c
@@ -102,6 +102,23 @@
 
 #include "clinic/_posixsubprocess.c.h"
 
+extern int python_wasm_fork_exec(char *const exec_array[],
+             char *const argv[],
+             char *const envp[],
+             const char *cwd,
+             int p2cread, int p2cwrite,
+             int c2pread, int c2pwrite,
+             int errread, int errwrite,
+             int errpipe_read, int errpipe_write,
+             int close_fds, int restore_signals,
+             int call_setsid, pid_t pgid_to_set,
+             int call_setgid, gid_t gid,
+             int call_setgroups, size_t groups_size, const gid_t *groups,
+             int call_setuid, uid_t uid,
+             int child_umask,
+             const void *child_sigmask,
+             int *py_fds_to_keep);
+
 /* Convert ASCII to a positive int, no libc call. no overflow. -1 on error. */
 static int
 _pos_int_from_ascii(const char *name)
@@ -1222,6 +1239,32 @@
         goto cleanup;
     }
 
+    int *cowasm_fds_to_keep = PyMem_Malloc((fds_to_keep_len + 1) * sizeof(int));
+    if (cowasm_fds_to_keep == NULL) {
+        PyErr_SetString(PyExc_MemoryError, "failed to malloc cowasm_fds_to_keep");
+        goto cleanup;
+    }
+    for (Py_ssize_t i = 0; i < fds_to_keep_len; i++) {
+        cowasm_fds_to_keep[i] = c_fds_to_keep[i];
+    }
+    cowasm_fds_to_keep[fds_to_keep_len] = -1;
+
+    pid = python_wasm_fork_exec(exec_array, argv, envp, cwd,
+                                p2cread, p2cwrite, c2pread, c2pwrite,
+                                errread, errwrite, errpipe_read, errpipe_write,
+                                close_fds, restore_signals, call_setsid, pgid_to_set,
+                                gid != (gid_t)-1, gid,
+                                extra_group_size >= 0,
+                                extra_group_size < 0 ? 0 : (size_t)extra_group_size,
+                                extra_groups,
+                                uid != (uid_t)-1, uid, child_umask,
+                                NULL, cowasm_fds_to_keep);
+    PyMem_Free(cowasm_fds_to_keep);
+    if (pid == (pid_t)-1) {
+        saved_errno = errno;
+    }
+    goto cleanup;
+
     /* This must be the last thing done before fork() because we do not
      * want to call PyOS_BeforeFork() if there is any chance of another
      * error leading to the cleanup: code without calling fork(). */
