--- native/Lib/test//test_unicode_file_functions.py	2022-08-05 07:45:18.000000000 -0700
+++ wasm/Lib/test//test_unicode_file_functions.py	2022-09-13 11:51:39.000000000 -0700
@@ -123,6 +123,9 @@
     # NFKD in Python is useless, because darwin will normalize it later and so
     # open(), os.stat(), etc. don't raise any exception.
     @unittest.skipIf(sys.platform == 'darwin', 'irrelevant test on Mac OS X')
+    # We also skip on wasi since the host platform might be darwin.  This test
+    # of course fails when the host is darwin and not when it isn't.
+    @unittest.skipIf(sys.platform == 'wasi', 'irrelevant test on Mac OS X')
     def test_normalize(self):
         files = set(self.files)
         others = set()
