diff --git a/Python/ceval.c b/Python/ceval.c
--- a/Python/ceval.c
+++ b/Python/ceval.c
@@ -380,7 +380,7 @@
 #  define Py_C_STACK_SIZE 4000000
 #endif
 
-#if defined(__EMSCRIPTEN__)
+#if defined(__EMSCRIPTEN__) && !defined(__wasi__)
 
// Temporary workaround to make `pthread_getattr_np` work on Emscripten.
// Emscripten 4.0.6 will contain a fix:
diff --git a/Include/internal/pycore_emscripten_signal.h b/Include/internal/pycore_emscripten_signal.h
--- a/Include/internal/pycore_emscripten_signal.h
+++ b/Include/internal/pycore_emscripten_signal.h
@@ -17,8 +17,8 @@
 #define _Py_CHECK_EMSCRIPTEN_SIGNALS_PERIODICALLY() _Py_CheckEmscriptenSignalsPeriodically()
 
-extern int Py_EMSCRIPTEN_SIGNAL_HANDLING;
-extern int _Py_emscripten_signal_clock;
+extern __attribute__((visibility("hidden"))) int Py_EMSCRIPTEN_SIGNAL_HANDLING;
+extern __attribute__((visibility("hidden"))) int _Py_emscripten_signal_clock;
 
 #else
 
diff --git a/Python/emscripten_signal.c b/Python/emscripten_signal.c
--- a/Python/emscripten_signal.c
+++ b/Python/emscripten_signal.c
@@ -24,7 +24,7 @@
     }
 });
 
-EMSCRIPTEN_KEEPALIVE int Py_EMSCRIPTEN_SIGNAL_HANDLING = 0;
+EMSCRIPTEN_KEEPALIVE __attribute__((visibility("hidden"))) int Py_EMSCRIPTEN_SIGNAL_HANDLING = 0;
 
 void
 _Py_CheckEmscriptenSignals(void)
@@ -39,7 +39,7 @@
 }
 
 #define PY_EMSCRIPTEN_SIGNAL_INTERVAL 50
-int _Py_emscripten_signal_clock = PY_EMSCRIPTEN_SIGNAL_INTERVAL;
+__attribute__((visibility("hidden"))) int _Py_emscripten_signal_clock = PY_EMSCRIPTEN_SIGNAL_INTERVAL;
 
 void
 _Py_CheckEmscriptenSignalsPeriodically(void)
