--- a/Python/ceval.c
+++ b/Python/ceval.c
@@ -380,7 +380,7 @@
 #  define Py_C_STACK_SIZE 4000000
 #endif

-#if defined(__EMSCRIPTEN__)
+#if defined(__EMSCRIPTEN__) && !defined(__wasi__)

 // Temporary workaround to make `pthread_getattr_np` work on Emscripten.
 // Emscripten 4.0.6 will contain a fix:
--- a/Include/internal/pycore_emscripten_signal.h
+++ b/Include/internal/pycore_emscripten_signal.h
@@ -17,8 +17,8 @@

 #define _Py_CHECK_EMSCRIPTEN_SIGNALS_PERIODICALLY() _Py_CheckEmscriptenSignalsPeriodically()
 
-extern int Py_EMSCRIPTEN_SIGNAL_HANDLING;
-extern int _Py_emscripten_signal_clock;
+extern __attribute__((visibility("hidden"))) int Py_EMSCRIPTEN_SIGNAL_HANDLING;
+extern __attribute__((visibility("hidden"))) int _Py_emscripten_signal_clock;
 
 #else
 
--- a/Python/emscripten_signal.c
+++ b/Python/emscripten_signal.c
@@ -24,7 +24,7 @@
     }
 });
 
-EMSCRIPTEN_KEEPALIVE int Py_EMSCRIPTEN_SIGNAL_HANDLING = 0;
+EMSCRIPTEN_KEEPALIVE __attribute__((visibility("hidden"))) int Py_EMSCRIPTEN_SIGNAL_HANDLING = 0;
 
 void
 _Py_CheckEmscriptenSignals(void)
@@ -39,7 +39,7 @@
 }
 
 #define PY_EMSCRIPTEN_SIGNAL_INTERVAL 50
-int _Py_emscripten_signal_clock = PY_EMSCRIPTEN_SIGNAL_INTERVAL;
+__attribute__((visibility("hidden"))) int _Py_emscripten_signal_clock = PY_EMSCRIPTEN_SIGNAL_INTERVAL;
 
 void
 _Py_CheckEmscriptenSignalsPeriodically(void)
--- a/configure.ac
+++ b/configure.ac
@@ -5182,6 +5182,10 @@
 AS_CASE([$ac_sys_system],
   [Emscripten], [
-    AS_VAR_APPEND([PLATFORM_OBJS], [' Python/emscripten_signal.o Python/emscripten_trampoline.o Python/emscripten_trampoline_wasm.o Python/emscripten_syscalls.o'])
+    AS_VAR_APPEND([PLATFORM_OBJS], [' Python/emscripten_signal.o Python/emscripten_trampoline.o Python/emscripten_trampoline_wasm.o'])
+    AS_CASE([$host],
+      [wasm32-unknown-wasi], [],
+      [AS_VAR_APPEND([PLATFORM_OBJS], [' Python/emscripten_syscalls.o'])]
+    )
     AS_VAR_APPEND([PLATFORM_HEADERS], [' $(srcdir)/Include/internal/pycore_emscripten_signal.h $(srcdir)/Include/internal/pycore_emscripten_trampoline.h'])
   ],
 )
--- a/configure
+++ b/configure
@@ -19199,7 +19199,13 @@
 case $ac_sys_system in #(
   Emscripten) :
-
-    as_fn_append PLATFORM_OBJS ' Python/emscripten_signal.o Python/emscripten_trampoline.o Python/emscripten_trampoline_wasm.o Python/emscripten_syscalls.o'
+    as_fn_append PLATFORM_OBJS ' Python/emscripten_signal.o Python/emscripten_trampoline.o Python/emscripten_trampoline_wasm.o'
+    case $host in #(
+  wasm32-unknown-wasi) :
+     ;; #(
+  *) :
+    as_fn_append PLATFORM_OBJS ' Python/emscripten_syscalls.o'
+     ;;
+esac
     as_fn_append PLATFORM_HEADERS ' $(srcdir)/Include/internal/pycore_emscripten_signal.h $(srcdir)/Include/internal/pycore_emscripten_trampoline.h'
    ;; #(
   *) :
