--- native/Modules/timemodule.c
+++ wasm/Modules/timemodule.c
@@ -381,7 +381,7 @@
 #ifdef _Py_MEMORY_SANITIZER
     __msan_unpoison(&clk_id, sizeof(clk_id));
 #endif
-    return PyLong_FromLong(clk_id);
+    return PyLong_FromLong((long)clk_id);
 }
 
 PyDoc_STRVAR(pthread_getcpuclockid_doc,
@@ -2029,67 +2029,67 @@
     if (HAVE_CLOCK_GETTIME_RUNTIME) {
 
 #ifdef CLOCK_REALTIME
-        if (PyModule_AddIntMacro(module, CLOCK_REALTIME) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_REALTIME", (long)CLOCK_REALTIME) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_MONOTONIC
-        if (PyModule_AddIntMacro(module, CLOCK_MONOTONIC) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_MONOTONIC", (long)CLOCK_MONOTONIC) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_MONOTONIC_RAW
-        if (PyModule_AddIntMacro(module, CLOCK_MONOTONIC_RAW) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_MONOTONIC_RAW", (long)CLOCK_MONOTONIC_RAW) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_HIGHRES
-        if (PyModule_AddIntMacro(module, CLOCK_HIGHRES) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_HIGHRES", (long)CLOCK_HIGHRES) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_PROCESS_CPUTIME_ID
-        if (PyModule_AddIntMacro(module, CLOCK_PROCESS_CPUTIME_ID) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_PROCESS_CPUTIME_ID", (long)CLOCK_PROCESS_CPUTIME_ID) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_THREAD_CPUTIME_ID
-        if (PyModule_AddIntMacro(module, CLOCK_THREAD_CPUTIME_ID) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_THREAD_CPUTIME_ID", (long)CLOCK_THREAD_CPUTIME_ID) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_PROF
-        if (PyModule_AddIntMacro(module, CLOCK_PROF) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_PROF", (long)CLOCK_PROF) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_BOOTTIME
-        if (PyModule_AddIntMacro(module, CLOCK_BOOTTIME) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_BOOTTIME", (long)CLOCK_BOOTTIME) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_TAI
-        if (PyModule_AddIntMacro(module, CLOCK_TAI) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_TAI", (long)CLOCK_TAI) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_UPTIME
-        if (PyModule_AddIntMacro(module, CLOCK_UPTIME) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_UPTIME", (long)CLOCK_UPTIME) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_UPTIME_RAW
-        if (PyModule_AddIntMacro(module, CLOCK_UPTIME_RAW) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_UPTIME_RAW", (long)CLOCK_UPTIME_RAW) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_MONOTONIC_RAW_APPROX
-        if (PyModule_AddIntMacro(module, CLOCK_MONOTONIC_RAW_APPROX) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_MONOTONIC_RAW_APPROX", (long)CLOCK_MONOTONIC_RAW_APPROX) < 0) {
             return -1;
         }
 #endif
 #ifdef CLOCK_UPTIME_RAW_APPROX
-        if (PyModule_AddIntMacro(module, CLOCK_UPTIME_RAW_APPROX) < 0) {
+        if (PyModule_AddIntConstant(module, "CLOCK_UPTIME_RAW_APPROX", (long)CLOCK_UPTIME_RAW_APPROX) < 0) {
             return -1;
         }
 #endif
