--- a/Include/internal/pycore_object.h
+++ b/Include/internal/pycore_object.h
@@ -111,8 +111,12 @@
+#ifdef __wasi__
+#  define _PyReftracerTrack(obj, operation) do { } while(0)
+#else
 #define _PyReftracerTrack(obj, operation) \
     do { \
         struct _reftracer_runtime_state *tracer = &_PyRuntime.ref_tracer; \
         if (tracer->tracer_func != NULL) { \
             void *data = tracer->tracer_data; \
             tracer->tracer_func((obj), (operation), data); \
         } \
     } while(0)
+#endif
