--- a/Objects/tupleobject.c
+++ b/Objects/tupleobject.c
@@ -961,6 +961,10 @@ _PyTuple_Resize(PyObject **pv, Py_ssize_t newsize)
     Py_ssize_t oldsize;
 
     v = (PyTupleObject *) *pv;
+    if (v != NULL && Py_IS_TYPE(v, &PyTuple_Type) && Py_SIZE(v) == newsize) {
+        return 0;
+    }
+
     if (v == NULL || !Py_IS_TYPE(v, &PyTuple_Type) ||
         (Py_SIZE(v) != 0 && !_PyObject_IsUniquelyReferenced(*pv))) {
         *pv = 0;
@@ -970,9 +974,6 @@ _PyTuple_Resize(PyObject **pv, Py_ssize_t newsize)
     }
 
     oldsize = Py_SIZE(v);
-    if (oldsize == newsize) {
-        return 0;
-    }
     if (newsize == 0) {
         Py_DECREF(v);
         *pv = tuple_get_empty();
