commit 48fec192ca77d2a8a874f973f469d4c16b21f034
Author: yperbasis <andrey.ashikhmin@gmail.com>
Date:   Wed Dec 16 11:15:35 2020 +0100

    Add longlong generic ABI

diff --git a/configure b/configure
index c27d6a8..ee29fa7 100755
--- a/configure
+++ b/configure
@@ -4078,7 +4078,9 @@ echo "define_not_for_expansion(\`HAVE_HOST_CPU_$tmp_host')" >> $gmp_tmpconfigm4p
 
 
 # abilist needs to be non-empty, "standard" is just a generic name here
-abilist="standard"
+abilist="longlong standard"
+# longlong ABI means mp_limb_t is (unsigned) long long
+limb_longlong=longlong
 
 # FIXME: We'd like to prefer an ANSI compiler, perhaps by preferring
 # c89 over cc here.  But note that on HP-UX c89 provides a castrated
