--- native/src/shell.c.in	2022-09-05 04:02:23.000000000 -0700
+++ wasm/src/shell.c.in	2022-10-17 19:19:08.000000000 -0700
@@ -234,7 +234,8 @@
 ** WebAssembly (WASM) bundle and need to disable and rewire a few
 ** things.
 */
-#ifdef __EMSCRIPTEN__
+// NO: for CoWasm we are disabling this *toy* sqlite shell mode.
+#ifdef __DISABLED_EMSCRIPTEN__
 #define SQLITE_SHELL_WASM_MODE
 #else
 #undef SQLITE_SHELL_WASM_MODE
