--- native/test/modules	2021-02-13 08:33:43.000000000 -0800
+++ wasm/test/modules	2022-10-23 17:19:10.000000000 -0700
@@ -83,7 +83,7 @@
 lrtest		progs		$(srcdir)	$(HEADER_DEPS)
 move_field	progs		$(srcdir)	$(HEADER_DEPS)	$(srcdir)/edit_field.h
 movewindow	progs		$(srcdir)	$(HEADER_DEPS)	$(srcdir)/popup_msg.h
-ncurses		progs		$(srcdir)	$(HEADER_DEPS)	$(incdir)/panel.h $(incdir)/menu.h $(incdir)/form.h
+#ncurses		progs		$(srcdir)	$(HEADER_DEPS)	$(incdir)/panel.h $(incdir)/menu.h $(incdir)/form.h
 newdemo		progs		$(srcdir)	$(HEADER_DEPS)
 padview		progs		$(srcdir)	$(HEADER_DEPS)	$(srcdir)/widechars.h	$(srcdir)/popup_msg.h
 pair_content	progs		$(srcdir)	$(HEADER_DEPS)
--- native/test/programs	2021-03-13 12:37:44.000000000 -0800
+++ wasm/test/programs	2022-10-23 17:20:06.000000000 -0700
@@ -79,7 +79,7 @@
 lrtest		$(LDFLAGS_CURSES)	$(LOCAL_LIBS)	lrtest
 move_field	$(LDFLAGS_DEFAULT)	$(LOCAL_LIBS)	move_field edit_field popup_msg
 movewindow	$(LDFLAGS_CURSES)	$(LOCAL_LIBS)	movewindow popup_msg
-ncurses		$(LDFLAGS_DEFAULT)	$(LOCAL_LIBS)	ncurses
+#ncurses		$(LDFLAGS_DEFAULT)	$(LOCAL_LIBS)	ncurses
 newdemo		$(LDFLAGS_CURSES)	$(LOCAL_LIBS)	newdemo
 padview		$(LDFLAGS_CURSES)	$(LOCAL_LIBS)	padview popup_msg
 pair_content	$(LDFLAGS_CURSES)	$(LOCAL_LIBS)	pair_content
