--- dist/native/lib/zig/compiler_rt.zig	2022-10-27 05:04:28.000000000 +0000
+++ dist.0/native/lib/zig/compiler_rt.zig	2022-10-21 04:38:54.000000000 +0000
@@ -212,6 +212,6 @@
     _ = @import("compiler_rt/memcpy.zig");
     _ = @import("compiler_rt/memset.zig");
     _ = @import("compiler_rt/memmove.zig");
-    _ = @import("compiler_rt/memcmp.zig");
+    _ = @import("compiler_rt/cmp.zig");
     _ = @import("compiler_rt/bcmp.zig");
 }
