--- native/numpy/core/setup.py	2022-09-09 06:36:30.000000000 -0700
+++ wasm/numpy/core/setup.py	2022-09-19 20:46:52.000000000 -0700
@@ -195,7 +195,7 @@
     check_funcs(OPTIONAL_STDFUNCS)
     check_funcs(OPTIONAL_FILE_FUNCS, headers=["feature_detection_stdio.h"])
     check_funcs(OPTIONAL_MISC_FUNCS, headers=["feature_detection_misc.h"])
-    
+
 
 
     for h in OPTIONAL_HEADERS:
@@ -266,6 +266,7 @@
 
     # Check for complex support
     st = config.check_header('complex.h')
+    st = False
     if st:
         priv.append(('HAVE_COMPLEX_H', 1))
         pub.append(('NPY_USE_C99_COMPLEX', 1))
