--- native/setup.py	2022-11-30 22:29:08
+++ wasm/setup.py	2022-11-30 22:29:56
@@ -22,6 +22,11 @@
     python -c 'import numpy; numpy.test()'
 
 """
+# COWASM -- we explicitly import setuptools first before anything else.  There's
+# a huge warning at startup if we don't do this, and then distutils gets messed up
+# with weird C++ compiler issues, etc.
+import setuptools
+
 DOCLINES = (__doc__ or '').split("\n")
 
 import os
