diff -urN a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -16,6 +16,7 @@
 fs = import('fs')

 is_windows = host_machine.system() == 'windows'
+is_wasi = host_machine.system() == 'wasi'

 # Additional targets
 py_with_pytest = py_module.find_installation(
diff -urN a/src/meson.build b/src/meson.build
--- a/src/meson.build
+++ b/src/meson.build
@@ -26,6 +26,28 @@
   numpy = declare_dependency(include_directories: inc_numpy)
 endif

+inc_numpy_cython = run_command(
+  py,
+  [
+    '-c',
+    '''
+from os.path import relpath
+from pathlib import Path
+import numpy
+path = Path(numpy.__file__).resolve()
+for parent in [path.parent, *path.parents]:
+  if parent.name == 'numpy':
+    path = parent
+    break
+try:
+  print(relpath(path))
+except Exception:
+  print(path)
+    '''.strip(),
+  ],
+  check: true,
+).stdout().strip()
+
 inc_cysignals = run_command(
   py,
   [
@@ -61,6 +61,24 @@
   check: true,
 ).stdout().strip()
 gmpy2 = declare_dependency(include_directories: inc_gmpy2)
+
+inc_memory_allocator = run_command(
+  py,
+  [
+    '-c',
+    '''
+from os.path import relpath
+import memory_allocator
+path = memory_allocator.__file__.replace('__init__.py', '')
+try:
+  print(relpath(path))
+except Exception:
+  print(path)
+    '''.strip(),
+  ],
+  check: true,
+).stdout().strip()
+
 gmp = dependency('gmp')

 if is_windows
@@ -103,7 +103,7 @@
   add_project_arguments('-I' + cypari2_cython_include, language: 'cython')
   # Cannot be found via pkg-config
   # We need to link against libpari directly to avoid runtime issues, so use pari as dependency of cypari2
-  pari = cc.find_library('pari', required: not is_windows, disabler: true)
+  pari = cc.find_library('pari', required: not is_windows and not is_wasi, disabler: true)
   if pari.found() and not meson.is_cross_build()
     # Verify PARI version
     pari_version_code = '''
@@ -178,7 +178,7 @@
 gd = dependency('gdlib', required: false, version: '>=2.1')
 if not gd.found()
   # Doesn't have a pkg-config file on some systems (https://github.com/conda-forge/libgd-feedstock/issues/55)
-  gd = cc.find_library('gd', required: not is_windows, disabler: true)
+  gd = cc.find_library('gd', required: not is_windows and not is_wasi, disabler: true)
 endif
 # Only some platforms have a standalone math library (https://mesonbuild.com/howtox.html#add-math-library-lm-portably)
 m = cc.find_library('m', required: false)
@@ -197,7 +197,7 @@
 m4rie = dependency('m4rie', required: false)
 if not m4rie.found()
   # For some reason, m4rie is not found via pkg-config on some systems (eg Conda)
-  m4rie = cc.find_library('m4rie', required: not is_windows, disabler: true)
+  m4rie = cc.find_library('m4rie', required: not is_windows and not is_wasi, disabler: true)
 endif
 mtx = dependency('libmtx', required: false)
 if not mtx.found()
@@ -220,26 +220,29 @@
   required: get_option('eclib'),
   disabler: true,
 )
-ecm = cc.find_library('ecm', required: not is_windows, disabler: true)
-gmpxx = dependency('gmpxx', required: not is_windows, disabler: true)
+ecm = cc.find_library('ecm', required: not is_windows and not is_wasi, disabler: true)
+gmpxx = dependency('gmpxx', required: not is_windows and not is_wasi, disabler: true)
 fflas = dependency(
   'fflas-ffpack',
-  required: not is_windows,
+  required: not is_windows and not is_wasi,
   disabler: true,
   version: '>=2.5.0',
 )
 givaro = dependency(
   'givaro',
-  required: not is_windows,
+  required: not is_windows and not is_wasi,
   disabler: true,
   version: '>=4.2.0',
 )
 linbox = dependency('linbox', required: false, version: '>=1.7.0')
 if not linbox.found()
-  linbox = cc.find_library('linbox', required: not is_windows, disabler: true)
+  linbox = cc.find_library('linbox', required: not is_windows and not is_wasi, disabler: true)
+endif
+mpc = cc.find_library('mpc', required: not is_windows and not is_wasi, disabler: true)
+mpfi = dependency('mpfi', required: false)
+if not mpfi.found()
+  mpfi = cc.find_library('mpfi', required: false)
 endif
-mpc = cc.find_library('mpc', required: not is_windows, disabler: true)
-mpfi = cc.find_library('mpfi', required: false)
 if not mpfi.found()
   mpfi_proj = subproject('mpfi')
   mpfi = mpfi_proj.get_variable('mpfi_dep')
@@ -252,15 +255,15 @@
   gap = cc.find_library(
     'gap',
     has_headers: ['gap/common.h'],
-    required: not is_windows,
+    required: not is_windows and not is_wasi,
     disabler: true,
   )
 endif
-singular = dependency('Singular', required: not is_windows, disabler: true)
+singular = dependency('Singular', required: not is_windows and not is_wasi, disabler: true)
 singular_factory = disabler()
 if singular.found()
   singular_factory = singular
-else
+elif not is_wasi
   singular_proj = subproject('singular')
   singular_factory = singular_proj.get_variable('factory_dep')
 endif
@@ -274,7 +277,7 @@
     ntl_lib_dir = candidate_ntl_lib
   endif
 endif
-ntl = disabler()
+ntl = dependency('ntl', required: false)
 if ntl_lib_dir != '' and cpp.has_header('NTL/config.h', args: ['-I' + ntl_include_dir], required: false)
   ntl = cpp.find_library(
     'ntl',
@@ -285,7 +288,7 @@
 endif
 if not ntl.found()
   # Cannot be found via pkg-config
-  ntl = cc.find_library('ntl', required: not is_windows, disabler: true)
+  ntl = cc.find_library('ntl', required: not is_windows and not is_wasi, disabler: true)
 endif

 boost_cpp17_compatible = false
@@ -314,6 +349,12 @@
 # https://github.com/mesonbuild/meson/issues/9562
 add_project_arguments('-I', meson.current_source_dir(), language: 'cython')
 add_project_arguments('-I', meson.current_build_dir(), language: 'cython')
+add_project_arguments('-I', inc_numpy_cython, language: 'cython')
+add_project_arguments('-I', join_paths(inc_numpy_cython, '..'), language: 'cython')
+add_project_arguments('-I', join_paths(inc_cysignals, '..'), language: 'cython')
+add_project_arguments('-I', inc_gmpy2, language: 'cython')
+add_project_arguments('-I', inc_memory_allocator, language: 'cython')
+add_project_arguments('-I', join_paths(inc_memory_allocator, '..'), language: 'cython')

 # Add global compiler flags
 add_project_arguments('-X auto_pickle=False', language: 'cython')
diff -urN a/src/sage/rings/integer.pyx b/src/sage/rings/integer.pyx
--- a/src/sage/rings/integer.pyx
+++ b/src/sage/rings/integer.pyx
@@ -312,7 +312,7 @@
 mpz_init(PARI_PSEUDOPRIME_LIMIT)
 mpz_ui_pow_ui(PARI_PSEUDOPRIME_LIMIT, 2, 64)

-cdef Parent the_integer_ring = integer_ring.ZZ
+cdef Parent the_integer_ring = None


 cdef inline Integer as_Integer(x):
@@ -7819,25 +7819,36 @@

 # zero and one initialization
 initialized = False
+cdef Integer zero = None
+cdef Integer one = None
+cdef list small_pool = []
+
+DEF small_pool_min = -5
+DEF small_pool_max = 256
+
+def init_integer_ring(parent):
+    global the_integer_ring
+    the_integer_ring = parent
+    set_zero_one_elements()
+
 cdef set_zero_one_elements():
-    global the_integer_ring, initialized
+    global the_integer_ring, initialized, zero, one, small_pool
     if initialized:
         return
+    if the_integer_ring is None:
+        return
     the_integer_ring._zero_element = Integer(0)
     the_integer_ring._one_element = Integer(1)
+    zero = the_integer_ring._zero_element
+    one = the_integer_ring._one_element
+    small_pool = [Integer(k) for k in range(small_pool_min, small_pool_max+1)]
     initialized = True
 set_zero_one_elements()

-cdef Integer zero = the_integer_ring._zero_element
-cdef Integer one = the_integer_ring._one_element
-
 # pool of small integer for fast sign computation
 # Use the same defaults as Python3 documented at
 # https://docs.python.org/3/c-api/long.html#c.PyLong_FromLong
-DEF small_pool_min = -5
-DEF small_pool_max = 256
 # we could use the above zero and one here
-cdef list small_pool = [Integer(k) for k in range(small_pool_min, small_pool_max+1)]

 cdef inline Integer smallInteger(long value):
     """
diff -urN a/src/sage/rings/integer_ring.pyx b/src/sage/rings/integer_ring.pyx
--- a/src/sage/rings/integer_ring.pyx
+++ b/src/sage/rings/integer_ring.pyx
@@ -1562,6 +1562,9 @@
         return self(python_int)

 ZZ = IntegerRing_class()
+import sage.rings.integer as _integer_module
+_integer_module.init_integer_ring(ZZ)
+del _integer_module
 Z = ZZ


diff -urN a/src/sage/rings/rational.pyx b/src/sage/rings/rational.pyx
--- a/src/sage/rings/rational.pyx
+++ b/src/sage/rings/rational.pyx
@@ -65,7 +65,13 @@
 import operator
 import fractions

-from sage.rings.rational_field import QQ
+QQ = None
+
+def init_rational_field(parent):
+    global QQ, the_rational_ring
+    QQ = parent
+    the_rational_ring = parent
+    set_zero_one_elements()

 from sage.arith.long cimport integer_check_long_py
 from sage.categories.morphism cimport Morphism
@@ -188,10 +194,14 @@

-cdef Parent the_rational_ring = QQ
+cdef Parent the_rational_ring = None

 # make sure zero/one elements are set
+initialized = False
 cdef set_zero_one_elements():
-    global the_rational_ring
+    global the_rational_ring, initialized
+    if initialized or the_rational_ring is None:
+        return
     the_rational_ring._zero_element = Rational(0)
     the_rational_ring._one_element = Rational(1)
+    initialized = True

 set_zero_one_elements()
diff -urN a/src/sage/rings/rational_field.py b/src/sage/rings/rational_field.py
--- a/src/sage/rings/rational_field.py
+++ b/src/sage/rings/rational_field.py
@@ -53,5 +53,11 @@
 """

+import sys
+_integer_ring_module = sys.modules.get('sage.rings.integer_ring')
+if _integer_ring_module is None or hasattr(_integer_ring_module, 'ZZ'):
+    from sage.rings.integer_ring import ZZ as _integer_ring_ZZ
+    del _integer_ring_ZZ
+del _integer_ring_module
 from sage.rings.integer import Integer
 from sage.rings.rational import Rational

@@ -1664,6 +1666,9 @@


 QQ = RationalField()
+import sage.rings.rational as _rational_module
+_rational_module.init_rational_field(QQ)
+del _rational_module
 Q = QQ


diff -urN a/src/sage/graphs/meson.build b/src/sage/graphs/meson.build
--- a/src/sage/graphs/meson.build
+++ b/src/sage/graphs/meson.build
@@ -9,7 +9,7 @@
 cliquer = dependency('libcliquer', required: false, disabler: true)
 if not cliquer.found()
   # fallback for older versions without pkg-config
-  cliquer = cc.find_library('cliquer', required: not is_windows, disabler: true)
+  cliquer = cc.find_library('cliquer', required: not is_windows and not is_wasi, disabler: true)
 endif

 planarity = dependency('libplanarity', required: false, disabler: true)
@@ -17,7 +17,7 @@
 if not planarity.found()
   # fallback for older versions without pkg-config
   planarity = cc.find_library(
     'planarity',
-    required: not is_windows,
+    required: not is_windows and not is_wasi,
     disabler: true,
   )
 endif
diff -urN a/src/sage/libs/all.py b/src/sage/libs/all.py
--- a/src/sage/libs/all.py
+++ b/src/sage/libs/all.py
@@ -2,7 +2,4 @@
-import sage.libs.ntl.all as ntl
-
-from sage.libs.pari.all import pari, pari_gen, PariError
-
-import sage.libs.symmetrica.all as symmetrica
-
 from sage.misc.lazy_import import lazy_import
+lazy_import('sage.libs.ntl', 'all', 'ntl')
+lazy_import('sage.libs.pari.all', ['pari', 'pari_gen', 'PariError'])
+lazy_import('sage.libs.symmetrica', 'all', 'symmetrica')
diff -urN a/src/sage/all.py b/src/sage/all.py
--- a/src/sage/all.py
+++ b/src/sage/all.py
@@ -211,4 +211,8 @@
     Suppress C-level startup chatter from optional runtimes in wheel installs.
     """
     def __enter__(self):
+        if sys.platform == 'wasi':
+            self._active = False
+            return
+
         import ctypes
diff -urN a/src/sage/rings/all.py b/src/sage/rings/all.py
--- a/src/sage/rings/all.py
+++ b/src/sage/rings/all.py
@@ -12,5 +12,6 @@
 # ****************************************************************************
 from sage.misc.lazy_import import lazy_import
+import sys

 # Ring base classes
 from sage.rings.ring import (Ring, Field, CommutativeRing, IntegralDomain,
@@ -51,6 +52,13 @@
 from sage.rings.finite_rings.all import *

 # Number field
-from sage.rings.number_field.all import *
+if sys.platform == 'wasi':
+    lazy_import('sage.rings.number_field.number_field',
+                ['NumberField', 'NumberFieldTower', 'CyclotomicField',
+                 'QuadraticField', 'is_real_place'])
+    lazy_import('sage.rings.number_field.number_field_element',
+                'NumberFieldElement')
+else:
+    from sage.rings.number_field.all import *

 # Function field
 from sage.rings.function_field.all import *
diff -urN a/src/sage/rings/finite_rings/finite_field_constructor.py b/src/sage/rings/finite_rings/finite_field_constructor.py
--- a/src/sage/rings/finite_rings/finite_field_constructor.py
+++ b/src/sage/rings/finite_rings/finite_field_constructor.py
@@ -175,4 +175,5 @@
 from collections import defaultdict
+import sys
 from sage.structure.category_object import normalize_names, certify_names
 from sage.rings.polynomial.polynomial_element import Polynomial
 from sage.rings.integer import Integer
@@ -180,12 +181,16 @@
-try:
-    # We don't late import this because this means trouble with the Givaro library
-    # On a Macbook Pro OSX 10.5.8, this manifests as a Bus Error on exiting Sage.
-    # TODO: figure out why
-    from .finite_field_givaro import FiniteField_givaro
-except ImportError:
+if sys.platform == 'wasi':
     FiniteField_givaro = None
-
-try:
-    from .finite_field_ntl_gf2e import FiniteField_ntl_gf2e
-except ImportError:
     FiniteField_ntl_gf2e = None
+else:
+    try:
+        # We don't late import this because this means trouble with the Givaro library
+        # On a Macbook Pro OSX 10.5.8, this manifests as a Bus Error on exiting Sage.
+        # TODO: figure out why
+        from .finite_field_givaro import FiniteField_givaro
+    except ImportError:
+        FiniteField_givaro = None
+
+    try:
+        from .finite_field_ntl_gf2e import FiniteField_ntl_gf2e
+    except ImportError:
+        FiniteField_ntl_gf2e = None
diff -urN a/src/sage/modular/arithgroup/meson.build b/src/sage/modular/arithgroup/meson.build
--- a/src/sage/modular/arithgroup/meson.build
+++ b/src/sage/modular/arithgroup/meson.build
@@ -51,4 +51,6 @@
     '-I',
     join_paths(meson.current_source_dir(), '../../../'),
+    '-I',
+    join_paths(inc_cysignals, '..'),
   ],
 )
diff -urN a/src/sage/libs/lcalc/meson.build b/src/sage/libs/lcalc/meson.build
--- a/src/sage/libs/lcalc/meson.build
+++ b/src/sage/libs/lcalc/meson.build
@@ -1,7 +1,7 @@
 lcalc = dependency(
   'lcalc',
   version: '>= 2.0.0',
-  required: not is_windows,
+  required: not is_windows and not is_wasi,
   disabler: true,
 )

diff -urN a/src/sage/libs/meson.build b/src/sage/libs/meson.build
--- a/src/sage/libs/meson.build
+++ b/src/sage/libs/meson.build
@@ -9,7 +9,7 @@
 endif
 # cannot be found via pkg-config
 ecl = cc.find_library('ecl', required: false, disabler: true)
-if not ecl.found() and not is_windows
+if not ecl.found() and not is_windows and not is_wasi
   ecl_proj = subproject('ecl')
   ecl = ecl_proj.get_variable('ecl_dep')
 endif
@@ -51,7 +51,7 @@
   maxima_present = false
 endif
 if not maxima_present
-  if is_windows
+  if is_windows or is_wasi
     # We currently cannot build Maxima on Windows, so we disable it
     maxima = disabler()
   else
@@ -110,7 +110,7 @@
 gc = dependency(
   ['bdw-gc-threaded', 'bdw-gc'],
   version: '>=7.6.4',
-  required: not is_windows,
+  required: not is_windows and not is_wasi,
   disabler: true,
 )
 homfly = dependency('libhomfly', required: false, disabler: true)
diff -urN a/src/sage/libs/symmetrica/meson.build b/src/sage/libs/symmetrica/meson.build
--- a/src/sage/libs/symmetrica/meson.build
+++ b/src/sage/libs/symmetrica/meson.build
@@ -3,7 +3,7 @@
   # fallback for older versions without pkg-config
   symmetrica = cc.find_library(
     'symmetrica',
-    required: not is_windows,
+    required: not is_windows and not is_wasi,
     disabler: true,
   )
 endif
diff -urN a/src/sage/matrix/meson.build b/src/sage/matrix/meson.build
--- a/src/sage/matrix/meson.build
+++ b/src/sage/matrix/meson.build
@@ -1,4 +1,4 @@
-iml = cc.find_library('iml', required: not is_windows, disabler: true)
+iml = cc.find_library('iml', required: not is_windows and not is_wasi, disabler: true)


 py.install_sources(
diff -urN a/src/sage/meson.build b/src/sage/meson.build
--- a/src/sage/meson.build
+++ b/src/sage/meson.build
@@ -43,7 +43,7 @@
   conf_data.set('OPENMP_CFLAGS', '-fopenmp')
   conf_data.set('OPENMP_CXXFLAGS', '-fopenmp')
 endif
-gap_exe = find_program('gap', required: not is_windows, disabler: true)
+gap_exe = find_program('gap', required: not is_windows and not is_wasi, disabler: true)
 if gap_exe.found()
   gaprun = run_command(
     gap_exe,
@@ -60,7 +60,7 @@
 endif
 ecm_bin = find_program(
   ['ecm', 'gmp-ecm'],
-  required: not is_windows,
+  required: not is_windows and not is_wasi,
   disabler: true,
 )
 conf_data.set('SAGE_ECMBIN', ecm_bin.full_path())
diff -urN a/src/sage/numerical/backends/meson.build b/src/sage/numerical/backends/meson.build
--- a/src/sage/numerical/backends/meson.build
+++ b/src/sage/numerical/backends/meson.build
@@ -1,5 +1,5 @@
 # Cannot be found via pkg-config
-glpk = cc.find_library('glpk', required: not is_windows, disabler: true)
+glpk = cc.find_library('glpk', required: not is_windows and not is_wasi, disabler: true)

 py.install_sources(
   '__init__.py',
diff -urN a/src/sage/rings/all.py b/src/sage/rings/all.py
--- a/src/sage/rings/all.py
+++ b/src/sage/rings/all.py
@@ -58,2 +58,17 @@
-from sage.rings.padics.all import *
-from sage.rings.padics.padic_printing import _printer_defaults as padic_printing
+if sys.platform == 'wasi':
+    lazy_import('sage.rings.padics.factory',
+                ['Zp', 'Zq', 'ZpCR', 'ZpCA', 'ZpFM', 'ZpFP', 'ZpLC',
+                 'ZpLF', 'ZqCR', 'ZqCA', 'ZqFM', 'ZqFP', 'ZpER',
+                 'Qp', 'Qq', 'QpCR', 'QpFP', 'QpLC', 'QpLF', 'QqCR',
+                 'QqFP', 'QpER', 'pAdicExtension'])
+    lazy_import('sage.rings.padics.factory', 'Zp', as_='pAdicRing')
+    lazy_import('sage.rings.padics.factory', 'Qp', as_='pAdicField')
+    lazy_import('sage.rings.padics.padic_generic', 'local_print_mode')
+    lazy_import('sage.rings.padics.pow_computer', 'PowComputer')
+    lazy_import('sage.rings.padics.pow_computer_ext', 'PowComputer_ext_maker')
+    lazy_import('sage.rings.padics.witt_vector_ring', 'WittVectorRing')
+    lazy_import('sage.rings.padics.padic_printing', '_printer_defaults',
+                as_='padic_printing')
+else:
+    from sage.rings.padics.all import *
+    from sage.rings.padics.padic_printing import _printer_defaults as padic_printing
diff -urN a/src/sage/rings/polynomial/polynomial_ring_constructor.py b/src/sage/rings/polynomial/polynomial_ring_constructor.py
--- a/src/sage/rings/polynomial/polynomial_ring_constructor.py
+++ b/src/sage/rings/polynomial/polynomial_ring_constructor.py
@@ -24,10 +24,18 @@
 from sage.structure.category_object import normalize_names
-
-try:
-    from sage.rings.padics import padic_base_leaves
-except ImportError:
+import sys
+
+if sys.platform == 'wasi':
     class padic_base_leaves:
         pAdicFieldCappedRelative = ()
         pAdicRingCappedRelative = ()
         pAdicRingCappedAbsolute = ()
         pAdicRingFixedMod = ()
+else:
+    try:
+        from sage.rings.padics import padic_base_leaves
+    except ImportError:
+        class padic_base_leaves:
+            pAdicFieldCappedRelative = ()
+            pAdicRingCappedRelative = ()
+            pAdicRingCappedAbsolute = ()
+            pAdicRingFixedMod = ()
